app/lib/features/calendar_sync/calendar_sync.dart

```diff
@@ -0,0 +1,283 @@
+    _logError(foundEventsResult, 'Failed to load calendar events') ?? [],
+  );
+
+  final newLinks = {};
+  final foundEventIds = [];
+  for (final eventAndRsvp in events) {
+    final calEvent = eventAndRsvp.event;
+    final rsvp = eventAndRsvp.rsvp;
+    final calEventId = calEvent.eventId().toString();
+    Event? localEvent;
+    if (currentLinks.containsKey(calEventId)) {
+      final localId = currentLinks[calEventId];
+      localEvent = foundEvents.cast<Event?>().firstWhere(
+            (e) => e?.eventId == localId,
+            orElse: () => null,
+          );
+    }
+
+    if (localEvent == null) {
+      localEvent = Event(calendarId);
+    } else {
+      foundEventIds.add(localEvent.eventId);
+    }
+
+    localEvent = await _updateEventDetails(calEvent, rsvp, localEvent);
+    final localRequest = await deviceCalendar.createOrUpdateEvent(localEvent);
+    if (localRequest == null) {
+      _log.severe('Updating $calEventId failed. No response. skipping');
+      continue;
+    }
+    final resultData = _logError(localRequest, 'Updating $calEventId failed');
+    if (resultData != null) {
+      newLinks[calEventId] = resultData;
+    } else {
+      _log.warning('Updating $calEventId failed. no new id given');
+      if (localEvent.eventId != null) {
+        // assuming that all went fine...
+        // maybe this is usual?
+        newLinks[calEventId] = localEvent.eventId;
+      }
+    }
+  }
+  final newMapping =
+      newLinks.entries.map((m) => '${m.key}=${m.value}').toList();
+  _log.info('Storing new mapping: $newMapping');
+  // set our new mapping
+  await preferences.setStringList(
+    calendarSyncIdsKey,
+    newMapping,
+  );
+
+  // time to clean up events that we aren't tracking anymore
+  for (final toDelete in foundEvents
+      .where((e) => e.eventId != null && !foundEventIds.contains(e.eventId))) {
+    _log.info('Deleting event ${toDelete.eventId}');
+    _logError(
+      await deviceCalendar.deleteEvent(calendarId, toDelete.eventId),
+      'Deleting local event $toDelete failed',
+    );
+  }
+}
+
+Future<Event> _updateEventDetails(
+  CalendarEvent acterEvent,
+  String? rsvp,
+  Event localEvent,
+) async {
+  localEvent.title = acterEvent.title();
+  localEvent.description = acterEvent.description()?.body();
+  localEvent.reminders = [Reminder(minutes: 10)];
+  localEvent.start = TZDateTime.from(
+    toDartDatetime(acterEvent.utcStart()),
+    UTC,
+  );
+  localEvent.end = TZDateTime.from(
+    toDartDatetime(acterEvent.utcEnd()),
+    UTC,
+  );
+  localEvent.status = switch (rsvp) {
+    'yes' => EventStatus.Confirmed,
+    'maybe' => EventStatus.Tentative,
+    _ => EventStatus.None
+  };
+  return localEvent;
+}
+
+Future<List<String>> _findActerCalendars() async {
+  // confirm this key exists.
+  final calendars = _logError(
+    await deviceCalendar.retrieveCalendars(),
+    'Failed to load calendars',
+  );
+  if (calendars == null) {
+    return [];
+  }
+  if (Platform.isAndroid) {
```

Comment:
Any reason for this platform specific condition?

Reply:
had to update this slightly for iOS support. But nothing major. Also limiting our deleting to _local_ calendars so we don't accidentally delete others ... like one they might have named "Acter" in their google calendar.